devlib/target.py

```diff
@@ -153,6 +153,15 @@ def __init__(self,
                  ):
         self.connection_settings = connection_settings or {}
         self.platform = platform or Platform()
+        if 'platform' in self.connection_settings:
+            if connection_settings['platform'] is not self.platform:
```

Comment:
There is an edge case where this fails. If the user instantiates like so:

    target = LinuxTarget(connection_settings={'platform': Platform()})

This will result in an error, because if a platform is not passed to the Target by the user, a default one will be automatically created (line 155); so if one was passed in the connection settings, there would still be two target objects.

line 155 should be updated to

    self.platform = connection_settings.get('platform', Platform())

so that the default platform is only created if one has not been passed to the connection settings.